blaze-client/src/main/scala/org/http4s/client/blaze/BlazeClient.scala

```diff
@@ -13,7 +13,19 @@ import scalaz.{-\/, \/-}
 object BlazeClient {
   private[this] val logger = getLogger
 
-  def apply[A <: BlazeConnection](manager: ConnectionManager[A], idleTimeout: Duration, requestTimeout: Duration): Client = {
+  def apply[A <: BlazeConnection](manager: ConnectionManager[A], config: BlazeClientConfig): Client = {
+
+    val shutdownTask = manager.shutdown().flatMap(_ => Task.delay {
```

Comment:
There isn't a lot to this class other than connecting the pipes; the `ConnectionManager` is really the core concept. Maybe the solution to your itch is to move this logic to the orbit of the `ConnectionManager` and then shutting it down will feel more natural to you?